File: Challenge/POLIGONO/poligono.dart
enum TipoPoligono { triangulo, cuadrado, rectangulo }

double calcularAreaPoligono(TipoPoligono tipo, List<double> args) {
  switch (tipo) {
    case TipoPoligono.triangulo:
      if (args.length != 2) {
        throw ArgumentError(
            'El triángulo requiere 2 argumentos: base y altura.');
      }
      double base = args[0];
      double altura = args[1];
      double area = 0.5 * base * altura;
      print("Área del triángulo: $area");
      return area;

    case TipoPoligono.cuadrado:
      if (args.length != 1) {
        throw ArgumentError('El cuadrado requiere 1 argumento: lado.');
      }
      double lado = args[0];
      double area = lado * lado;
      print("Área del cuadrado: $area");
      return area;

    case TipoPoligono.rectangulo:
      if (args.length != 2) {
        throw ArgumentError(
            'El rectángulo requiere 2 argumentos: base y altura.');
      }
      double base = args[0];
      double altura = args[1];
      double area = base * altura;
      print("Área del rectángulo: $area");
      return area;

    default:
      throw ArgumentError('Tipo de polígono no soportado.');
  }
}

void main() {
  calcularAreaPoligono(TipoPoligono.triangulo, [2, 3]);
  calcularAreaPoligono(TipoPoligono.cuadrado, [2]);
  calcularAreaPoligono(TipoPoligono.rectangulo, [2, 3]);
}

File: Challenge/POLIGONO/poligono.py
def calcular_area_poligono(tipo, *args):
    # *args es una tupla con los argumentos que se pasan a la función

    # tipo es el tipo de polígono
    if tipo.lower() == "triangulo" and len(args) == 2: # len(args) es el número de argumentos
        base, altura = args
        area = 0.5 * base * altura
        print(f"Área del triángulo: {area}")
        return area
    elif tipo.lower() == "cuadrado" and len(args) == 1: # len(args) es el número de argumentos
        lado = args[0]
        area = lado * lado
        print(f"Área del cuadrado: {area}")
        return area
    elif tipo.lower() == "rectangulo" and len(args) == 2: # len(args) es el número de argumentos
        base, altura = args
        area = base * altura
        print(f"Área del rectángulo: {area}")
        return area
    else:
        print("Tipo de polígono no soportado o número incorrecto de argumentos.")
        return None # Si no se cumple ninguna de las condiciones anteriores, se retorna None

# Ejemplos de uso
calcular_area_poligono("triangulo", 4, 6)
calcular_area_poligono("cuadrado", 5)
calcular_area_poligono("rectangulo", 3, 8)
